pkg/authenticator/ecrsecret.go

```diff
@@ -0,0 +1,110 @@
+package authenticator
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"os"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	varPackagesNamespace = "eksa-packages"
+	varConfigMapName     = "installed-namespaces"
+	varConfigMapKey      = "NAMESPACES"
+	varECRTokenName      = "ecr-token"
+)
+
+type ecrSecret struct {
+	clientset kubernetes.Interface
+}
+
+var _ Authenticator = (*ecrSecret)(nil)
+
+func NewECRSecret(config *rest.Config) *ecrSecret {
+	clientset, _ := kubernetes.NewForConfig(config)
+
+	return &ecrSecret{
+		clientset: clientset,
+	}
+}
+
+func (s *ecrSecret) AuthFilename() (string, error) {
+	// Check if Helm registry config is set
+	helmconfig := os.Getenv("HELM_REGISTRY_CONFIG")
+	if helmconfig != "" {
+		// Use HELM_REGISTRY_CONFIG
+		return helmconfig, nil
+	}
+
+	return "", nil
+}
+
+func (s *ecrSecret) UpdateConfigMap(ctx context.Context, namespace string, add bool) error {
+	cm, err := s.clientset.CoreV1().ConfigMaps(varPackagesNamespace).
+		Get(ctx, varConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	addToConfigMap(cm, namespace, add)
+	_, err = s.clientset.CoreV1().ConfigMaps(varPackagesNamespace).
+		Update(ctx, cm, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func addToConfigMap(cm *v1.ConfigMap, namespace string, add bool) {
+	if namespace == "" {
+		namespace = "default"
```

Comment:
So if a namespace isn't specified we "default" is what we add to the configmap not `eksa-packages` just trying to clarify

Reply:
Yes, though this is changed now to expect a defaultNamespace and use "default" if that is not set